File: wails-app/inventory-management-updated/frontend/src/types/square.ts
// Square OAuth and configuration types

export interface SquareConfig {
  application_id: string;
  environment: 'production' | 'sandbox';
  redirect_uri: string;
}

export interface SquareConfigCreate {
  application_id: string;
  application_secret: string;
  environment: 'production' | 'sandbox';
  redirect_uri?: string;
}

export interface SquareOAuthToken {
  merchant_id: string;
  token_type: string;
  scopes: string;
  created_at: string;
  updated_at: string;
  has_token: boolean;
  expires_at?: string;
}

export interface SquareAuthorizationURLResponse {
  authorization_url: string;
  scopes: string[];
  state?: string;
}

export interface SquareOAuthStatus {
  isConfigured: boolean;
  isAuthorized: boolean;
  merchantId?: string;
  config?: SquareConfig;
  token?: SquareOAuthToken;
}

File: wails-app/inventory-management-updated/pkg/domain/square.go
package domain

import "time"

// SquareOAuthToken represents the OAuth token for a Square merchant
type SquareOAuthToken struct {
	TokenID      int64     `json:"token_id"`
	MerchantID   string    `json:"merchant_id"`
	AccessToken  string    `json:"access_token"`
	RefreshToken string    `json:"refresh_token,omitempty"`
	ExpiresAt    *time.Time `json:"expires_at,omitempty"`
	TokenType    string    `json:"token_type"`
	Scopes       string    `json:"scopes"` // Comma-separated list of scopes
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

// SquareConfig represents Square application configuration
// These will be stored in the Settings table with keys prefixed with "square_"
type SquareConfig struct {
	ApplicationID     string `json:"application_id"`
	ApplicationSecret string `json:"application_secret"`
	Environment       string `json:"environment"` // "sandbox" or "production"
	RedirectURI       string `json:"redirect_uri"`
}
